Public/Src/Engine/Processes/ExternalToolSandboxedProcess.cs

```diff
@@ -0,0 +1,239 @@
+﻿// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+using System;
+using System.Collections.Generic;
+using System.ComponentModel;
+using System.Diagnostics;
+using System.Diagnostics.ContractsLight;
+using System.Text;
+using System.Threading.Tasks;
+using BuildXL.Interop;
+using BuildXL.Native.IO;
+using BuildXL.Utilities.Tasks;
+
+namespace BuildXL.Processes
+{
+    /// <summary>
+    /// Sanboxed process that will be executed by an external tool.
+    /// </summary>
+    public class ExternalToolSandboxedProcess : ExternalSandboxedProcess

```

Comment:
This class looks very similar to `UnSandboxedProcess`.  Did you consider reusing `UnSandboxedProcess`?

Reply:
Yes, I was about to reuse UnSandboxedProcess, but there are a couple of items that need to be refactored. For example, in this class we're using StringBuilder, while in the other class we are using SandboxedProcessOutputBuilder. Also, this class needs the tool path, and is not callable from SandboxedProcessFactory. 

I'll keep them separate for now. But in the future, I may want to merge them; perhaps by creating a more general class that both this class and UnSandboxedProcess can use.

Another consideration is this is a subclass of `ExternalSandboxedProcess`. Another subclass that will be created soon is the one that send a command to VM to execute the sandboxed process executor tool.

---
In reply to: [276462691](https://github.com/Microsoft/BuildXL/pull/159#discussion_r276462691) [](ancestors = 276462691)

Reply:
I decided to refactor the common part into AsyncProcessExecutor.

---
In reply to: [276465387](https://github.com/Microsoft/BuildXL/pull/159#discussion_r276465387) [](ancestors = 276465387,276462691)